src/Microsoft.AspNetCore.Blazor/Rendering/ComponentState.cs

```diff
@@ -15,6 +15,7 @@ namespace Microsoft.AspNetCore.Blazor.Rendering
     internal class ComponentState
     {
         private readonly int _componentId; // TODO: Change the type to 'long' when the Mono runtime has more complete support for passing longs in .NET->JS calls
+        private readonly ComponentState _parentComponentState;
```

Comment:
Tracking this new data is the main result of this PR. We don't do anything with it yet (hence the artificialness of the unit test coverage), but later on it will be key to having the renderer wire up the deep params.